src/main/java/seedu/address/logic/commands/EditApplicantCommand.java

```diff
@@ -0,0 +1,151 @@
+package seedu.address.logic.commands;
+
+import static java.lang.Integer.parseInt;
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_APPLICANT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_APPLICANT_WITH_ID;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.applicant.Applicant;
+import seedu.address.model.listing.Listing;
+
+/**
+ * Edits a listing in the listing book.
+ */
+public class EditApplicantCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit_app";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an applicant's details for a listing. "
+            + "by the index number of the listing used in the displayed listing book. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_APPLICANT_WITH_ID + "OLD_APPLICANT]...\n"
```

Comment:
This usage make it sound like u can only edit applicants who have their id displayed. id should be optional. also iirc the [] is for optional parameters but old and new applicant are compulsory parameters

Reply:
<img width="651" alt="image" src="https://user-images.githubusercontent.com/88370247/227132934-998dd3a5-0ce6-45e8-84a3-f8218d2ee1a8.png">

Reply:
the PREFIX_APPLICANT_WITH_ID is just the id/ flag. we use that for deletion also so i put as that if not is abit confusing. ill change the square brackets thing.